### PR TITLE
Fix case sensitivity issue and pagination params issue when filtering with groups in the SCIM2 Users endpoint

### PR DESCRIPTION
**Pull request by Yasara-Y**
Below improvements are included in this fix:

- Fix the issue on filtering users with hybrid roles when pagination params are present.
- Fix the case sensitivity issue when filtering with groups in the SCIM2 Users endpoint.

### CODE REVIEW

**Review by vihanga-liyanage — APPROVED** [review]

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -1515,6 +1515,35 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
...
+    private Set<org.wso2.carbon.user.core.common.User> filterUsersByGroup(Node node, int offset, int limit,
+                                                                          String domainName) throws CharonException {
+
+        // Set filter values.
+        String attributeName = ((ExpressionNode) node).getAttributeValue();
+        String filterOperation = ((ExpressionNode) node).getOperation();
+        String attributeValue = ((ExpressionNode) node).getValue();
+
+        // If there is a domain, append the domain with the domain separator in front of the new attribute value if
+        // domain separator is not found in the attribute value.
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1527]: The multi-line comments should follow the following format.
```
/*
 *
 *
 */
```

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -1515,6 +1515,35 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
...
+        }
+
+        try {
+            List<String> roleNames = getRoleNames(attributeName, filterOperation, attributeValue);
+            Set<org.wso2.carbon.user.core.common.User> users = getUserListOfRoles(roleNames);
+            users = paginateUsers(users, limit, offset);
+            return users;
+        } catch (UserStoreException e) {
+            String errorMessage = String.format("Error while filtering the users for filter with attribute name: "
+                            + "%s, filter operation: %s and attribute value: %s. ", attributeName, filterOperation,
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1541]: Extra space at the end of the comment.

Comment (somindatommy) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1541]: Extra space at the end of the comment.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -1515,6 +1515,35 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
...
+        // Set filter values.
+        String attributeName = ((ExpressionNode) node).getAttributeValue();
+        String filterOperation = ((ExpressionNode) node).getOperation();
+        String attributeValue = ((ExpressionNode) node).getValue();
+
+        // If there is a domain, append the domain with the domain separator in front of the new attribute value if
+        // domain separator is not found in the attribute value.
+        if (StringUtils.isNotEmpty(domainName) && StringUtils
+                .containsNone(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
+            attributeValue = domainName.toUpperCase() + CarbonConstants.DOMAIN_SEPARATOR +
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1530]: Can use `UserCoreUtil.addDomainToName` method to do this.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 1 above)
Comment (Yasara-Y) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1527]: Resolved with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/395

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 3 above)
Comment (Yasara-Y) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1530]: Resolved with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/395

**Pull request merged**